server/handlers/default.go

```diff
@@ -211,6 +214,65 @@ func getKeyHandler(ctx context.Context, w http.ResponseWriter, r *http.Request,
 	return nil
 }
 
+// ListVersionsHandler serves a paginated list of versions of a given TUF role
+func ListVersionsHandler(ctx context.Context, w http.ResponseWriter, r *http.Request) error {
+	vars := mux.Vars(r)
+	gun, ok := vars["imageName"]
+	if !ok || gun == "" {
+		return errors.ErrUnknown.WithDetail("no gun")
+	}
+	role, ok := vars["tufRole"]
+	if !ok || role == "" {
+		return errors.ErrUnknown.WithDetail("no role")
+	}
+
+	err := r.ParseForm()
+	if err != nil {
+		return errors.ErrUnknown.WithDetail(err)
+	}
+
+	start, number, err := parsePageParams(r.Form)
+	if err != nil {
+		return errors.ErrBadPagination.WithDetail(err)
+	}
+
+	s := ctx.Value("metaStore")
+	store, ok := s.(storage.MetaStore)
+	if !ok || store == nil {
+		return errors.ErrNoStorage.WithDetail("No storage configured")
+	}
+	return listVersionsHandler(ctx, w, store, gun, role, start, number)
+}
+
+func parsePageParams(f url.Values) (start string, number int, err error) {
+	start = f.Get("start")
```

Comment:
Non-blocking - for an HTTP API, unless we are trying to make this pagination more general (so it applies to more things), maybe more descriptive query parameters would make it easier to understand how it works?  Maybe something like `from_checksum` and `num_versions` or some such?  (I'm bad at naming, so probably something else :D)